lib/delayed/backend/base.rb

```diff
@@ -139,6 +139,10 @@ def max_run_time
         end
       end
 
+      def destroy_failed_jobs
+        payload_object.destroy_failed_jobs if payload_object.respond_to?(:destroy_failed_jobs)
+      end
+
```

Comment:
Then here, update the method to return `Delayed::Worker.destroy_failed_jobs` if the payload does not respond to `destroy_failed_jobs`